src/dagmc.cpp

```diff
@@ -72,6 +72,28 @@ DAGUniverse::DAGUniverse(pugi::xml_node node)
     adjust_material_ids_ = get_node_value_bool(node, "auto_mat_ids");
   }
 
+  // get material assignment overloading
+  if (check_for_node(node, "mat_assignment")) {
+    auto mat_node = node.child("mat_assignment");
+    // loop over all attributes (each attribute corresponds to a material)
+    for (pugi::xml_attribute attr = mat_node.first_attribute(); attr;
+         attr = attr.next_attribute()) {
+      // Store assignment reference name
+      std::string mat_ref_assignment = attr.name();
+
+      // Get mat name for each assignement instances
+      std::stringstream iss {attr.value()};
```

Comment:
Let's use the `split` function from `string_utils.h` here.

Reply:
A reorganization of the XML node might make this easier also. We can revisit that later.